testing/ivm_helper.ts

```diff
@@ -58,6 +58,7 @@ export async function setupIvmContext(bundlePath: string, executionContext: Exec
     authenticationName: executionContext.authenticationName,
     executionId: executionContext.executionId,
     previousAttemptError: executionContext.previousAttemptError,
+    syncState: executionContext.syncState,
```

Comment:
Can we maybe make a type guard `isSyncExecutionContext()` and do:

```ts
syncState: isSyncExecutionContext(executionContext) ? executionContext.syncState : undefined,
```

I know it seems like we always call this with a SyncExecutionContext but I think we should be more accurate so there's less to unwind later if we want to clean up the callers too.